packages/cli/src/setup-gcloud.ts

```diff
@@ -1,27 +1,102 @@
 import path from 'node:path';
-import { $, fs } from 'zx';
+import {
+  $,
+  fs,
+  question,
+  echo,
+} from 'zx';
 
+let accessTokenGCP: string | undefined;
 const serviceAccountId = 'cloud-commerce-gh-actions';
 const getAccountEmail = (projectId: string) => {
   return `${serviceAccountId}@${projectId}.iam.gserviceaccount.com`;
 };
 
-const checkServiceAccountExists = async (projectId: string) => {
+const requestApi = async (
+  projectId: string,
+  accessToken: string,
+  options?: {
+    baseURL?: string,
+    url?: string,
+    method: string,
+    body?: string,
+  },
+) => {
+  const body = options?.body;
+  let url = options?.baseURL || `https://iam.googleapis.com/v1/projects/${projectId}/serviceAccounts`;
+  url += options?.url || '';
+
+  const data = await (await fetch(
+    url,
+    {
+      method: options?.method || 'GET',
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+        'Content-Type': 'application/json; charset=utf-8',
+      },
+      body,
+    },
+  )).json() as any;
+  const { error } = data;
+  if (error) {
+    let msgErr = 'Unexpected error in request';
+    msgErr = error.message ? `code: ${error.code} - ${error.message}` : msgErr;
+    const err = new Error(msgErr);
+    throw err;
+  }
+
+  return data;
+};
+
+const getAcessTokenGCP = async () => {
+  await echo`-- Get the Google administrator account credentials.
+    (i) To obtain them, access (https://shell.cloud.google.com/?fromcloudshell=true&show=terminal)
+
+    (ii) Execute 'gcloud auth application-default print-access-token' in cloud shell`;
+
+  const accessToken = await question('accessToken: ');
+
+  return accessToken;
+};
+
+const checkServiceAccountExists = async (projectId: string, accessToken?: string) => {
```

Comment:
Mas por que raios adicionou o `accessToken` de argumento só dessa função e usou o global em todas as outras? 🙃

Reply:
Caramba eu to ficando é mais tatu 🤦‍♂️